### PR TITLE
ci: cilium dualstack for load test pipeline

### PR DESCRIPTION
**Pull request by camrynl**
**Reason for Change**:
- Creates a new stage for cilium dualstack in the load test pipeline
- Added new parameters to the load test template in order to reuse stages and detect dualstack scenarios
- Converted `cilium/1.14` daemonset and operator files from` .tpl` to `.yaml` for consistency and fluid parameterization 

**Issue Fixed**:

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] relevant PR labels added

**Notes**:

### CODE REVIEW

File: .pipelines/cni/cilium/cilium-overlay-load-test-template.yaml [hunk 1]
```diff
@@ -9,6 +9,8 @@ parameters:
   arch: ""
   osSKU: Ubuntu
   hubbleEnabled: false
+  dualstackVersion: ""
+  cni: ""
```
Comment (jpayne3506) [inline .pipelines/cni/cilium/cilium-overlay-load-test-template.yaml:13]: nit: can default this to `cilium` as this is a cilium template

File: .pipelines/cni/pipeline.yaml [hunk 2]
```diff
@@ -315,6 +319,17 @@ stages:
...
+  
+  - template: cilium/cilium-overlay-load-test-template.yaml
+    parameters:
+      name: cilium_overlay_ds
+      clusterType: dualstack-byocni-nokubeproxy-up
+      clusterName: "cilium-ds-ov"
+      nodeCount: ${NODE_COUNT_CILIUM}
+      vmSize: ${VM_SIZE_CILIUM}
+      dualstackVersion: ${CILIUM_DUALSTACK_VERSION}
+      cni: "cilium_dualstack"
```
Comment (jpayne3506) [inline .pipelines/cni/pipeline.yaml:332]: Is there a region dependency for cilium dualstack? This would impact cluster creation.

Comment (camrynl) [reply .pipelines/cni/pipeline.yaml:332]: We should be able to create clusters for the regions we test with. I was able to get the cluster created in the pipeline a few times last week

**Comment by jpayne3506** [comment]
I checked the CNI Release Test pipeline and did not see a passing Dualstack E2E. All other changes are 🟢

**Comment by vipul-21** [comment]
lgtm, can we get a green  on cni release pipeline ?

**Comment by camrynl** [comment]
I can send the run where the dualstack stage passes, but I would also like to gather a fully green run with the latest changes once the pipeline is back up

**Review by jpayne3506 — APPROVED** [review]

**Review by vipul-21 — APPROVED** [review]

**Pull request merged**